### PR TITLE
Do not attempt push to ghcr from forked pull requests

### PR DESCRIPTION
**Pull request by banesullivan**
Following up on https://github.com/pyvista/pyvista/pull/2030#issuecomment-1013508900

~@adeak, would you please make a pull request off of this branch to test this if you have a moment?~ I have a burner account that I am testing on right now in #2052 and #2053

### CODE REVIEW

**Comment by banesullivan** [comment]
No matter what, forked PR's cannot push to the ghcr (for good reason): https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

@adeak's  #2030 was from a fork. What I will do is is disable the action so that it only runs on PRs within this repo.

**Comment by banesullivan** [comment]
Should be good to go - tested in #2053

**Comment by banesullivan** [comment]
@adeak, FYI, you have the ability to push to branches directly in this repo - feel free to do so if it makes your life easier

**Comment by adeak** [comment]
> @adeak, FYI, you have the ability to push to branches directly in this repo - feel free to do so if it makes your life easier

Thanks @banesullivan, I'm aware :) But as far as I'm concerned I'm happy with not being able to muck up the main repo with botched pushes, my workflow is not hindered by the indirection. Of course if there's a technical reason why I should switch to branches here I can do that.

What exactly does this action do?

**Comment by banesullivan** [comment]
> What exactly does this action do?

This action was introduced in #2011: it builds the Dockerfile in `docker/Dockerfile` and bushes the image to this repositories container registry [here](https://github.com/pyvista/pyvista/pkgs/container/pyvista). There is a step in the action that handles adding meaningful labels to the image such that every PR in this repo is tagged as `pr-#`. This means we can easily pull the docker image for any PR in the repo to try out new features.

Considering we follow [a branch naming scheme](https://github.com/pyvista/pyvista/blob/main/CONTRIBUTING.md#branch-naming-conventions), we should probably ignore building/pushing on some branches. E.g., `junk/`, `maint/`, `doc/`, `no-ci/`. But perhaps this should be a separate PR that edits all relevant GH Actions.

**Comment by adeak** [comment]
Thanks for the explanation. So does this mean that since this action doesn't work for third-party repo branches, you can't do this docker pulling game for PRs from "regular" contributors (and myself if I keep using my fork as the base for PRs)?

And if yes, how problematic is this for you? Do you and @akaszynski (and maybe others?) regularly check out PRs like this? I'm mostly asking because this is the kind of technical reason I mentioned that would mean I should switch to using branches on this repo directly.

**Comment by banesullivan** [comment]
> So does this mean that since this action doesn't work for third-party repo branches, you can't do this docker pulling game for PRs from "regular" contributors (and myself if I keep using my fork as the base for PRs)?

That is correct, though there is a way to still share/publish the image by having the contributor enable actions for their fork. In which case we may need to change the `on:` condition for this action... I will investigate this a bit more and try to demo it in a junk PR from a personal fork.

> how problematic is this for you? Do you and @akaszynski (and maybe others?) regularly check out PRs like this?

For most things, honestly, I rely on passing tests and don't actually pull the changes to test myself (though I'm not heavily reviewing these days). However, for new features or things that might change the behavior of the API, I do like to pull the changes, and having this sort of Docker set up so I can easily pull the changes into a working environment is a huge time saver for me personally at least.

I think the main benefit of building/publishing these images is for two scenarios I see frequently: 

1. @akaszynski or I are working on some change to PyVista for a niche use case where we are deploying PyVista - in which case, being able to point to a new base Docker image will be a major convenience.
2. Users request a bug fix that is tough for us to reproduce without their data/use case. We can point users to this docker image when fixing a bug which can simplify their testing when they are not familiar with cloning the source.

**Comment by adeak** [comment]
> That is correct, though there is a way to still share/publish the image by having the contributor enable actions for their fork.

That sounds like too much effort, alas.

> I think the main benefit of building/publishing these images is for two scenarios I see frequently:
>     [...]

If I again understand correctly both cases might be satisfied by a "recent enough" PR opened by a member on this repo. In any case I digress; I think I can stick to PRs from my own repo until anyone tells me otherwise :)

**Review by akaszynski — APPROVED** [review]
Thanks for fixing this. I agree with @banesullivan that this has the potential to be useful for various projects, especially investigating test failures as the environment can be pulled without having to work through duplicating it locally.

One thing to consider is adding back in the doc requirements, even though this will lead to large image sizes.

**Comment by banesullivan** [comment]
> One thing to consider is adding back in the doc requirements, even though this will lead to large image sizes.

Let's leave that for a different PR. I'm thinking I will add a separate image for the doc examples (and perhaps bundle all of the example python files in that image too)

File: .github/workflows/docker-package.yml [hunk 1]
```diff
@@ -44,6 +43,6 @@ jobs:
         with:
           context: .
           file: docker/Dockerfile
-          push: true
+          push: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
```
Comment (banesullivan) [inline .github/workflows/docker-package.yml:46]: Wait... I just realized this might not push on `main` or releases....

Comment (banesullivan) [reply .github/workflows/docker-package.yml]: Fixed

**Comment by banesullivan** [comment]
Okay! NOW this is ready to go... tested in #2063 and confirmed working in https://github.com/banesullivan/localtileserver as well

**Pull request merged**